Create methods "translateAllBreedsByLinguatools", "createBreedTransla…

udodikdmitro opened the pull request:
…tion"

Commit:
Create methods "translateAllBreedsByLinguatools", "createBreedTranslation"

Commit:
Expanse breed translation column, add breed translation controller

Commit:
Add test, fix class BreedRepository

Commit:
Fix some methods
- translateAllBreedsByLinguatools() for returning messages
- testTranslateAllBreedsByLinguatools_should_save_only_absent_breed_translations_once() for correctly translation breeds
- controller translateAllBreedsByLinguatools() for printing exception message into browser

HailToSG commented on src/test/java/com/catapi/service/TranslationServiceTest.java:
```diff
@@ -198,4 +204,40 @@ private String getTranslationResponse(String testToTranslate) {
     }
+
+    @Test
+    void testTranslateAllBreedsByLinguatools_should_save_only_absent_breed_translations_once() {
+        BreedTranslation breedTranslation1Uk = new BreedTranslation();
+        breedTranslation1Uk.setBreedName("Порода1");
+        breedTranslation1Uk.setDescription("Опис1");
+        breedTranslation1Uk.setLocale(Locale.UK);
+        Breed breed1 = new Breed();
+        breed1.setId(1L);
+        breed1.setBreedName("Breed1");
+        breed1.setDescription("Description1");
+        breed1.setBreedTranslations(List.of(breedTranslation1Uk));
+
+        BreedTranslation breedTranslation2Ru = new BreedTranslation();
+        breedTranslation2Ru.setBreedName("Порода2");
+        breedTranslation2Ru.setDescription("Описание2");
+        breedTranslation2Ru.setLocale(Locale.RU);
+        Breed breed2 = new Breed();
+        breed2.setId(2L);
+        breed2.setBreedName("Breed2");
+        breed2.setDescription("Description2");
+        breed2.setBreedTranslations(List.of(breedTranslation2Ru));
+
+        when(breedRepository.findAll()).thenReturn(List.of(breed1, breed2));
+
+        mockOkHttpClientResponse("Breed2 @@@ Description2", "Порода2 @@@ Опис2");
+
+        translationService.translateAllBreedsByLinguatools(Locale.UK);
+
+        ArgumentCaptor<BreedTranslation> breedCaptor = ArgumentCaptor.forClass(BreedTranslation.class);
+        verify(breedTranslationRepository, times(1)).save(breedCaptor.capture());
+
+        BreedTranslation saveBreedTranslation = breedCaptor.getAllValues().getFirst();
+        assertEquals("Порода2", saveBreedTranslation.getBreedName());
+        assertEquals("Опис2", saveBreedTranslation.getDescription());
+    }
```
private methods should go after all the rest

udodikdmitro replied:
Moved

HailToSG commented on src/main/java/com/catapi/service/TranslationService.java:
```diff
@@ -121,4 +130,39 @@ private String extractTranslation(String scriptContent) {
         throw new ExternalApiException("Wrong response format");
     }
+
+    public String translateAllBreedsByLinguatools(Locale locale) {
```
You DON'T need to return String here

udodikdmitro replied:
Fixed

HailToSG commented on src/main/java/com/catapi/controller/TranslationController.java:
```diff
@@ -38,4 +42,22 @@ public void updateTranslationManually(@PathVariable Long catFactTranslationId, @
         return translationService.translate(locale, textToTranslate);
     }
+
+    @PostMapping("/linguatools/translateAllBreeds")
+    public ResponseEntity<String> translateAllBreedsByLinguatools(@RequestParam Locale locale) {
+        try {
+            if (EnumSet.of(Locale.UK, Locale.RU).contains(Locale.valueOf(locale.toString()))) {
+                return ResponseEntity.ok(translationService.translateAllBreedsByLinguatools(locale));
+            } else {
+                throw new IllegalArgumentException(STR."Can not get translation with using parameter \{locale}");
+            }
+        } catch (IllegalArgumentException e) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.getMessage());
+        }
+    }
```
You don't need to

1. return ResponseEntity<String>
2. check locale
3. put logic in controllers

udodikdmitro replied:
Fixed

Commit:
Fix some methods
- add new method for calling translateAllBreedsByLinguatools() and writing exceptions into the browser;
- move private methods into a class end;
- delete code logic and locale checking from controller

Commit:
Final fix
